starter-code/scripts/controllers/articleController.js

```diff
@@ -45,6 +48,7 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  // This method uses the articleView path, which takes all our article instances and renders them all to the page.
```

Comment:
A more detailed explanation: look carefully at the beginning of the articleController.loadByCategory function definition -- it is not setting ctx.articles to our articles array. It is simply declaring a function and assigning it to the variable articleData. Nothing is being called at this point, and the ctx object is not being altered until articleData is invoked. 

There is a crucial conditional statement present - if we have an already populated articles array, then we don't have to worry about doing anything more and we **can** simply add an articles property to the ctx object with the value of the Article.allArticles array. Then, the next function in the middleware chain is called. If we don't have any articles yet (on initial pageload), we will call Article.fetchAll and pass in the function that we defined, articleData, as a callback so that we can retrieve our articles from our JSON file via AJAX or from local storage, save a reference to them as Article.allArticles, and THEN call our callback (which attach them to the context object and calls the next function in the middleware chain).